pkg/machine/ocipull/pull.go

```diff
@@ -45,11 +55,12 @@ func Pull(ctx context.Context, imageInput types.ImageReference, localDestPath *d
 	}
 
 	path, err := policyPath()
-	if err != nil {
-		return err
+	if err == nil {
+		policy, err = signature.NewPolicyFromFile(path)
+	} else {
```

Comment:
This may mask permission errors. I think we should add a `errors.Is(err, os.ErrNotExist)` check.

Reply:
This will not work, the policyPath() function does not return a error that can be matched like this.

Re-reading the code here I am not sure this is doing what we want at all, policyPath() returns the DefaultPolicyJSONPath path without checking for existence because that was not needed because NewPolicyFromFile checked this.

Now this could would fail if MACHINE_POLICY_JSON_DIR was set at build time which likely is not the intend?

The best thing would be likely to change policyPath() to return the slice of all paths, then loop over it here with `NewPolicyFromFile(path)` and match the error from the with `errors.Is(err, fs.ErrNotExist)` and the after the loop check if polcy is still nil then fall back to the NewPolicyFromBytes() call.

Reply:
+1 @Luap99 IMO this is the right way to do it.